src/raft.c

```diff
@@ -465,6 +512,10 @@ RedisModuleCallReply *RaftExecuteCommandArray(RedisRaftCtx *rr,
         }
     }
 
+    if (is_multi_session) {
+        endClientSession(rr, client_session->client_id);
+    }
```

Comment:
Same as my above comment, can we send WATCH and then some other commands without MULTI? If so, we should not end the session if I'm not mistaken.